src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -1161,8 +1161,19 @@ await notificationService.PublishUpdateAsync(cr.ModelResource, s => s with
         }
     }
 
+    private async Task WaitForDependencyResourcesAsync(IResource resource, CancellationToken cancellationToken)
+    {
+        var resourceLogger = loggerService.GetLogger(resource);
+        var waitContext = new WaitContext(resourceLogger, serviceProvider);
+        var waitAnnotations = resource.Annotations.OfType<WaitAnnotation>().Select(w => w.WaitAsync(waitContext, cancellationToken)).ToArray();
+        Task.WaitAll(waitAnnotations, cancellationToken);
```

Comment:
Is this a mistake? This is blocking

Reply:
Not a mistake (at this point). This is the spot which enforces the wait to ensure other resources are healthy - but it highlights a problem with `ApplicationExecutor` that we have today which is that all the resources are started sequentially.

I think it would be good to discuss strategy here.

Reply:
Resources are not started sequentially.